ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -183,6 +191,84 @@ static void validateDefaultTopicFormats(final KsqlConfig config) {
     validateTopicFormat(config, KsqlConfig.KSQL_DEFAULT_VALUE_FORMAT_CONFIG, "value");
   }
 
+  private static void validateFips(final KsqlConfig config, final KsqlRestConfig restConfig) {
+    if (config.getBoolean(ConfluentConfigs.ENABLE_FIPS_CONFIG)) {
+      validateStateStoreType(restConfig);
+
+      final FipsValidator fipsValidator = ConfluentConfigs.buildFipsValidator();
+
+      final Map<String, List<String>> fipsTlsMap = new HashMap<>();
+      fipsTlsMap.put(KsqlRestConfig.SSL_CIPHER_SUITES_CONFIG,
+          restConfig.getList(KsqlRestConfig.SSL_CIPHER_SUITES_CONFIG));
+      fipsTlsMap.put(KsqlRestConfig.SSL_ENABLED_PROTOCOLS_CONFIG,
+          restConfig.getList(KsqlRestConfig.SSL_ENABLED_PROTOCOLS_CONFIG));
+
+      final Map<String, SecurityProtocol> securityProtocolMap = new HashMap<>();
+      final String brokerSecurityProtocol =
+          config.getString(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG);
+      securityProtocolMap.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+          SecurityProtocol.forName(brokerSecurityProtocol));
+
+      try {
+        // validate cipher suites and TLS version
+        fipsValidator.validateFipsTls(fipsTlsMap);
+
+        // validate broker
+        fipsValidator.validateFipsBrokerProtocol(securityProtocolMap);
+
+        // validate ssl endpoint algorithm
+        fipsValidator.validateRestProtocol(
+            restConfig.getString(KsqlConfig.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG));
+
+        // validate schema registry url
+        fipsValidator.validateRestProtocol(
+            determineProtocol(restConfig.getString(KsqlConfig.SCHEMA_REGISTRY_URL_PROPERTY)));
+
+        // validate all listeners
+        final List<String> listeners = restConfig.getList(KsqlRestConfig.LISTENERS_CONFIG);
+        for (String listener: listeners) {
+          fipsValidator.validateRestProtocol(determineProtocol(listener));
+        }
+        final List<String> proxyListeners =
+            restConfig.getList(KsqlRestConfig.PROXY_PROTOCOL_LISTENERS_CONFIG);
+        for (String listener: proxyListeners) {
+          fipsValidator.validateRestProtocol(determineProtocol(listener));
+        }
+        fipsValidator.validateRestProtocol(
+            determineProtocol(restConfig.getString(KsqlRestConfig.ADVERTISED_LISTENER_CONFIG)));
+        fipsValidator.validateRestProtocol(
+            determineProtocol(restConfig.getString(KsqlRestConfig.INTERNAL_LISTENER_CONFIG)));
+        log.info("FIPS mode enabled for ksqlDB!");
+      } catch (final Exception e) {
+        log.error("SSL configs violates FIPS!" + e.getMessage());
```

Comment:
will this msg be detailed enough about diagnosing which of the above validation steps failed?